SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestNetReference/SpatialTestNetReference.cpp

```diff
@@ -0,0 +1,207 @@
+
+			for (AActor* ArrayObject : CubesWithReferences)
+			{
+				ACubeWithReferences* CurrentCube = Cast<ACubeWithReferences>(ArrayObject);
+				FVector CurrentCubeLocation = CurrentCube->GetActorLocation();
+
+				bool bHasCorrectReferences = true;
+				int ValidReferences = 0;
+
+				for (AActor* OtherObject : CubesWithReferences)
+				{
+					ACubeWithReferences* OtherCube = Cast<ACubeWithReferences>(OtherObject);
+					FVector OtherCubeLocation = OtherObject->GetActorLocation();
+
+					// If the cube is the current one or the diagonally opposed one, then ignore it as it should never be a neigbhour of the current cube
+					if(OtherCubeLocation.Equals(CurrentCubeLocation) || (OtherCubeLocation.X == -CurrentCubeLocation.X && OtherCubeLocation.Y == -CurrentCubeLocation.Y))
+					{
+						continue;
+					}
+
+					// Check that the current cube has a neighbour reference to this OtherCube
+					bHasCorrectReferences &= (CurrentCube->Neighbour1 == OtherCube) || (CurrentCube->Neighbour2 == OtherCube);
+
+					if (bHasCorrectReferences)
+					{ 
+						++ValidReferences;
+					}
+				}
+
+				if (ValidReferences == 0)
+				{
+					// Check that the current cube has 0 valid references
+					bHasCorrectReferences &= !IsValid(CurrentCube->Neighbour1) && !IsValid(CurrentCube->Neighbour2);
+				}
+				else if (ValidReferences == 1)
+				{
+					// We have previously checked that one neighbour reference is correctly pointing to the neighbour cube, also check that the other reference is null
+					bHasCorrectReferences &=  !IsValid(CurrentCube->Neighbour1) || !IsValid(CurrentCube->Neighbour2);
+				}
+
+				Test->AssertEqual_Bool(bHasCorrectReferences, true, TEXT("References are correct"));
+			}
+
+			// Make the character move to the next step in the sequence
+			Test->LocationIndex++;
+
+			// After checking all locations twice, finish the test
+			if (Test->LocationIndex == 2 * Test->TestLocations.Num())
+			{
+				// Reset this variable in order to allow for the test to be run multiple times
+				Test->LocationIndex = 0;
+
+				Test->FinishStep();
+			}
+		}
+	});
+
+	AddServerStep(TEXT("SpatialTestNetReferenceServerCleanup"), 1, nullptr, nullptr, [](ASpatialFunctionalTest* NetTest, float DeltaTime) {
+		ASpatialTestNetReference* Test = Cast<ASpatialTestNetReference>(NetTest);
+
+		// Destroy the previously spawned cubes
+		TArray<AActor*> SpawnedObjects;
+		UGameplayStatics::GetAllActorsOfClass(Test->GetWorld(), ACubeWithReferences::StaticClass(), SpawnedObjects);
+
+		for (AActor* CubeToDestroy : SpawnedObjects)
+		{
+			CubeToDestroy->Destroy();
+		}
+
+		// Possess the original pawn, so that the spawned character can get destroyed correctly
+		Test->OriginalPawn.Key->Possess(Test->OriginalPawn.Value);
+
+		// Reset the PositionUpdateFrequency
+		GetMutableDefault<USpatialGDKSettings>()->PositionUpdateFrequency = Test->PreviousPositionUpdateFrequency;
```

Comment:
If the test fatals before this is hit does that mean the GDK settings are still modded?

Reply:
I do think so yes, completely missed this fact.